### ISSUE TITLE
Dealing with CONSTRUCT calls in sparql library

### ISSUE THREAD

**Issue by fils** [ec-geolink/glservices#1]
At present we have to do CONSTRUCT calls as simple http GET calls.  Can we do this in the sparql library itself. In the past I had issues doing things other than SELECT.

**Comment by krisnadhi** [comment]
What do you mean in "the SPARQL library”?

**Comment by fils** [comment]
The SPARQL library is at github.com/knakk/sparql.   It is a Go lang SPARQL library.  However, while it's very nice for SELECT calls I am not sure it can accommodate CONSTRUCT calls.    I've had to code around it.

**Comment by krisnadhi** [comment]
I have never used this before, but it seems to me that the results returned by Query is in the form of a set of bindings (i.e., map). On the other hand, CONSTRUCT calls are expected to returns a set of triples, which may not be easily formatted as bindings.

**Comment by fils** [comment]
agreed...   I don't think we are going to be able to use it for CONSTRUCT.   I already put in an issue over in it's repo.   What I have works. but I'd use a library if I could.

**Comment by fils** [comment]
The library only does SELECT calls.  So we will need to do DESCRIBE and CONSTRUCT ourselves unless we fell up to working on this library.  Worth a look around I guess, but I doubt we want to dive into that unless it looks real clean to do.   Ref:  https://github.com/knakk/sparql/issues/2#issuecomment-123211789

**Comment by ashepherd** [comment]
i'd be interested in seeing how far we got with adding CONSTRUCT/DESCRIBE support to the library. I'll play around with it this weekend if I can
